src/app/clusters/energy-evse-server/energy-evse-server.cpp

```diff
@@ -0,0 +1,438 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "energy-evse-server.h"
+
+#include <app/AttributeAccessInterface.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/util/attribute-storage.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::EnergyEvse;
+using namespace chip::app::Clusters::EnergyEvse::Attributes;
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace EnergyEvse {
+
+CHIP_ERROR Instance::Init()
+{
+    registerAttributeAccessOverride(this);
+    InteractionModelEngine::GetInstance()->RegisterCommandHandler(this);
+    return CHIP_NO_ERROR;
+}
+
+void Instance::Shutdown()
+{
+    unregisterAttributeAccessOverride(this);
+    InteractionModelEngine::GetInstance()->UnregisterCommandHandler(this);
+}
+
+bool Instance::HasFeature(Feature aFeature) const
+{
+    return mFeature.Has(aFeature);
+}
+
+bool Instance::SupportsOptAttr(OptionalAttributes aOptionalAttrs) const
+{
+    return mOptionalAttrs.Has(aOptionalAttrs);
+}
+
+bool Instance::SupportsOptCmd(OptionalCommands aOptionalCmds) const
+{
+    return mOptionalCmds.Has(aOptionalCmds);
+}
+
+// AttributeAccessInterface
+CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    switch (aPath.mAttributeId)
+    {
+    case State::Id:
+        return aEncoder.Encode(mDelegate.GetState());
+    case SupplyState::Id:
+        return aEncoder.Encode(mDelegate.GetSupplyState());
+    case FaultState::Id:
+        return aEncoder.Encode(mDelegate.GetFaultState());
+    case ChargingEnabledUntil::Id:
+        return aEncoder.Encode(mDelegate.GetChargingEnabledUntil());
+    case DischargingEnabledUntil::Id:
+        /* V2X */
+        if (!HasFeature(Feature::kV2x))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
```

Comment:
Chances are, the checks here for attributes are not needed, because if we don't claim to support the attributes (as we should not), this code will never be reached for those attribute IDs.

Reply:
Ok I'll back out these checks then.

Reply:
Reverted them out.